### ISSUE TITLE
Route53 Traffic policy record

### ISSUE THREAD

**Issue by perrfect** [StackExchange/dnscontrol#585]
Hi.
On Route53 i have Traffic policy for geolocation for a domain.
Can i manage this type of record through dnscontrol?

**Comment by tlimoncelli** [comment]
Hi!

DNSControl doesn't currently manage that.  We'd be open to PRs that add that feature.

I believe that Terraform supports that kind of thing.

**Comment by perrfect** [comment]
Thank you

**Comment by perrfect** [comment]
> @perrfect Did you find a solution for this?

Hello. No i didn't find the solution(

**Comment by truthdoug** [comment]
Adding some notes here because I got curious about this again. 

For this example, traffic from the EU goes to a cloudfront distribution and other traffic goes to an S3 bucket. 

Using the aws command line, I grabbed all json data for my zone like this:
 `aws route53 list-resource-record-sets --hosted-zone-id /hostedzone/redacted`

The geolocated records look like this:
```json
{
    "ResourceRecordSets": [
        {
            "Name": "example.org.",
            "Type": "A",
            "SetIdentifier": "default-configuration-geolocation",
            "GeoLocation": {
                "CountryCode": "*"
            },
            "AliasTarget": {
                "HostedZoneId": "redacted1",
                "DNSName": "s3-example-us-east-1.amazonaws.com.",
                "EvaluateTargetHealth": false
            }
        },
        {
            "Name": "example.org.",
            "Type": "A",
            "SetIdentifier": "europe-record",
            "GeoLocation": {
                "ContinentCode": "EU"
            },
            "AliasTarget": {
                "HostedZoneId": "redacted2",
                "DNSName": "example.cloudfront.net.",
                "EvaluateTargetHealth": false
            }
        }
    }
}
```

Using `dnscontrol get-zones -format=js`, the resulting lines currently look like this:

```js
R53_ALIAS('@', 'A', 's3-example-us-east-1.amazonaws.com.', R53_ZONE('redacted1')),
R53_ALIAS('@', 'A', 'example.cloudfront.net.', R53_ZONE('redacted2')),
```

From a UX point of view, my first thought was that the geolocation could be added as an argument to `R53_ALIAS()` but that won't work because other types of records (`A`, `CNAME`, etc) can also have geolocation variants.

I've looked a bit at `providers/route53/route53Provider.go` but I'm not sure where to start with this.

Is there a similar provider-specific feature that might be a good model for implementing this?

**Comment by tlimoncelli** [comment]
Ah!  That json is very enlightening!

My recommendation is to add something like: `R53_GEOZONE()` that would accept a dictionary with all the various parameters.  That would be stored in the metadata for that record for use by the provider.

```
R53_ALIAS('@', 'A', 's3-example-us-east-1.amazonaws.com.', R53_ZONE('redacted1'),
          R53_GEO( {
            "SetIdentifier": "default-configuration-geolocation",
            "GeoLocation": {
                "CountryCode": "*"
            },
            "AliasTarget": {
                "HostedZoneId": "redacted1",
                "DNSName": "s3-example-us-east-1.amazonaws.com.",
                "EvaluateTargetHealth": false
            }
       }),
R53_ALIAS('@', 'A', 'example.cloudfront.net.', R53_ZONE('redacted2'),
       R53_GEO( {
            "SetIdentifier": "europe-record",
            "GeoLocation": {
                "ContinentCode": "EU"
            },
            "AliasTarget": {
                "HostedZoneId": "redacted2",
                "DNSName": "example.cloudfront.net.",
                "EvaluateTargetHealth": false
            }
       }),
```

It isn't pretty, but ... we could add a R53_GEO_BUILDER() function later.

I lack time to write this kind of feature (Stack isn't giving me time to work on features that Stack doesn't use).  However I'd be glad to walk you through it.  It would be relatively straight forward.  The integration tests take out a lot of the guess-work.

**Comment by truthdoug** [comment]
I'm intrigued by the idea of writing support for this with your guidance, @tlimoncelli 

... in the interim, I just read about the new [IGNORE](https://docs.dnscontrol.org/language-reference/domain-modifiers/ignore) feature. Am I correct in thinking that this might be a good stop gap solution? By using `IGNORE()` on the records that have the Route53 geolocation designation, I could then manage the other records on this zone using dnscontrol. 

Does that sound right?

**Comment by tlimoncelli** [comment]
Yes!  That should work!  Try putting an IGNORE() in that matches any records you create manually (i.e. clickops)

**Comment by zozo** [comment]
@truthdoug Do you have any update on this topic?

**Comment by truthdoug** [comment]
@zozo I added `IGNORE` records for the geolocation records. It works well. When we need to update those, we do it directly through the AWS web console. We can manage all the rest of the records in dnscontrol

**Comment by zozo** [comment]
@truthdoug  So you don't manage geo-records with dnscontrol?
That's actually our main idea, since we are using many geo-records and want to manage them with dnscontrol.

**Comment by truthdoug** [comment]
@zozo Correct. Of a few hundred records that we manage with dnscontrol, we only have four that use geolocation and those change very infrequently.

@tlimoncelli offers [an implementation idea in a comment above](https://github.com/StackExchange/dnscontrol/issues/585#issuecomment-1535383842). It hasn't risen in importance for our use to spend enough time on this. If you were to explore it, I'd be willing to help test!

**Comment by zozo** [comment]
Yes, I checked all the conversation in this issue. I will definitely let you know if we will go that way. But at the moment seems like it would be easier for us to just generate terraform config to manage geo-records on route53/cloudns.
